plutus-core/plutus-ir/test/transform/inline/letFunConstMulti

```diff
@@ -0,0 +1,21 @@
+{-
+Multiple occurrences of the const function:
+The const function should not be inlined by the unconditional inlining optimization.
+The first constFun should be inlined the the optimization that inlines fully applied functions.
+The second constFun is not fully applied so it should not be inlined.
+ 
+let 
+  constFun :: Integer -> Integer -> Integer
+  constFun x y = x
+in constFun 3 (constFun 3) 
```

Comment:
This doesn't typecheck - `constFun 3` is not an integer

Reply:
Oops, I just found another advantage of writing in PlutusTx! ;)

Polymorphic `const` sounds interesting, maybe I should add that?